enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,316 @@
+
+Additionally, any error within the backup sidecar, the container goes into `CrashLoopBackOff`, which leads to the whole `Pod` reports `NotReady` status. Consequently, the whole `Etcd-cluster` can not server requests, nor the `API-Server`.
+
+
+#### DaemonSet manages an Etcd-backup pod within every control-plane node.
+
+This approach separates the `etcd-backup-server` deployment from the etcd static pods, to avoid affecting etcd-cluster and API-Server availability upon any error within the etcd-backup-server.
+
+The advantage of using `DaemonSet` over Deployment is to guarantee scheduling etcd-backup-server pods on all `Control-plane` nodes, regardless of the number of nodes (e.g. 1, 3, 5) using `nodeSelector` capability.
+By setting the `nodeSelector` to the master node label (i.e. `node-role.kubernetes.io/master`), etcd-backup-server's pods are guaranteed to scheduled on every control-plane node.
+
+This method has been implemented and tested. The same container used in the Sidecar approach has been utilised. However, the scheduling is being managed by a DaemonSet.
+Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by creating a DaemonSet resource, with the `etcd-backup-server` container and the configurations applied in the CR.
+During [testing](https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261), this method has shown a solid and stable deployment, where the Backup pods shown only `1` restart over a duration of `45` minutes. Moreover, the cause of the restart on all the Backup pods were due to `node` issue, not the backup container.
+
+_**Pros**_
+
+As the Backup pods are being scheduled independently of the etcd pods, any error within the Backup pods does not influence cluster Openshift cluster availability.
+
+Moreover, as the Backup pods are being scheduled within every master nodes, the backups are being taken from every local etcd member on the same node. This avoids overwhelming a specific member with backup requests.
+
+Since a Backup pod exist on every master node, and a `hostPath` volume is being used to store the backups, it is guaranteed to have available backups on all master nodes, which is required for disaster recovery scenarios.
+
+_**Cons**_
+
+No issue has been encountered during testing this approach.
+
+### Open Questions
+
+- What values could be used for default configurations, considering all OCP variants.
+- What is the minimum disk size needed in order to operate this feature without running out of disk space ?
+- How to handle an empty `schedule` ? 
+- How to mitigate losing backups upon control-plane scaling using `CPMSO` ?
+
```

Comment:
alternative approach we just discussed on the call:
* have a no-config backup controller inside of CEO running the delay queue
* at every interval create backup CRs for each control plane node

that way we have maximum reuse of all components. We have to add a node selector to our current backup CR however.

Reply:
To allow configuration of this option, could the `PeriodicBackup` still be used, but have a `hostPath` option that triggers this mode of operation.

Once concern with having no PeriodicBackup at the higher level would be that the user has no way to adjust the default backup configuration should they want to

Reply:
raised https://github.com/openshift/cluster-etcd-operator/pull/1363 yesterday to test the approach proposed above

- Used parallelism in JobSpec = 3 (i.e. control-plane nodes)
- Used `NodeAffinity` in order to schedule each `Job` on  master nodes only
- Used `PodAntiAffinity` in order to guarantee that each `Job` is being scheduled on a different master node

However, the approach did **not** work for the following reasons 

- But the Backup [CR](https://github.com/openshift/api/blob/684b2b1679baeb7742ec7b76c6eab8e0c8036e56/config/v1alpha1/types_backup.go#L99) expects a PVC per CR, this is not possible here, since we need 3 different PVCs, one per each node, before hand, and the CR expects only one PVC
- following a suggestion [here](https://github.com/openshift/cluster-etcd-operator/pull/1363#issuecomment-2444641024), I tried to create a `hostPath` volume within each `Jop.Spec` from within `PeriodicBackupController` to overcome the PVC singleton issue, however, the `request-backup` and `prune` containers that each `Backup` job consist of, expect `PVC` as an arg.

Reply:
And how is this different then to the daemonset approach? Is the daemonset approach using a different backing up binary?

Did the pods schedule as expected, even if they weren't able to proceed due to PVC mounting issues?

Reply:
> And how is this different then to the daemonset approach? Is the daemonset approach using a different backing up binary?

So the DS approach uses a separate binary (i.e. `etcd-backup-server`) see https://github.com/openshift/cluster-etcd-operator/pull/1354 

The `etcd-backup-server` expects all the `Backup` CR as args .. also it uses default path with `hostPath` volume to store backups 

> Did the pods schedule as expected, even if they weren't able to proceed due to PVC mounting issues?

So the pods were not scheduled since the PVC were not mounted, the pods could not become `Ready`